ci-operator/step-registry/ipi/conf/inframachineset/ipi-conf-inframachineset-commands.sh

```diff
@@ -0,0 +1,134 @@
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,infra]}
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/infra: ""
+EOF
+
+# Get machineset name to generate a generic template
+ref_machineset_name=$(oc get machinesets -n openshift-machine-api |grep worker |awk 'NR==1{print $1}')
+
+# Get machineset temaplte
+template_machineset=$(oc get machineset $ref_machineset_name -n openshift-machine-api -o json)
+
+# Replace machine name worker to infra
+infra_machineset_name=$(echo ${template_machineset} |jq -c '.metadata.name' |sed -e 's/worker/infra/' |sed -e 's/^"//' -e 's/"$//')
+
+export ref_machineset_name infra_machineset_name
+# Get a templated json from worker machineset, change machine type and machine name
+# and pass it to oc to create a new machine set
+oc get -nopenshift-machine-api machineset $ref_machineset_name -o json \
+    | jq --arg INFRA_NODE_TYPE "${INFRA_NODE_TYPE}" '.spec.template.spec.providerSpec.value.instanceType = $INFRA_NODE_TYPE' \
+    | jq --arg infra_machineset_name "${infra_machineset_name}" '.metadata.name = $infra_machineset_name' \
+    | jq --arg infra_machineset_name "${infra_machineset_name}" '.spec.template.metadata.labels."machine.openshift.io/cluster-api-machineset" = $infra_machineset_name' \
+    | jq --arg infra_machineset_name "${infra_machineset_name}" '.spec.selector.matchLabels."machine.openshift.io/cluster-api-machineset" = $infra_machineset_name' \
+    | jq --arg infra_machineset_name "${infra_machineset_name}" '.spec.template.spec.metadata.labels."node-role.kubernetes.io/infra" = ""' \
+    | jq 'del(.status)|del(.metadata.selfLink)|del(.metadata.uid)' \
+    | oc create -f -
+
+# Scale machineset to expected number of replicas
+oc -n openshift-machine-api scale machineset/"${infra_machineset_name}" --replicas="${INFRA_NODE_REPLICAS}"
+
+# Wait for infra nodes to come up
+while [[ $(oc -n openshift-machine-api get machineset/${infra_machineset_name} -o 'jsonpath={..status.readyReplicas}') != "${INFRA_NODE_REPLICAS}" ]]; do echo -n "." && sleep 5; done
+
+# Move router
+oc apply -f - <<EOF
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  nodePlacement:
+    nodeSelector:
+      matchLabels:
+        node-role.kubernetes.io/infra: ""
+EOF
+
+# Move registry
+oc apply -f - <<EOF
+apiVersion: imageregistry.operator.openshift.io/v1
+kind: Config
+metadata:
+  name: cluster
+spec:
+  affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - podAffinityTerm:
+          namespaces:
+          - openshift-image-registry
+          topologyKey: kubernetes.io/hostname
+        weight: 100
+  nodeSelector:
+    node-role.kubernetes.io/infra: ""
+EOF
+
+# Move monitoring pods
+oc apply -f - <<EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
```

Comment:
We have at least one other step that messes with this resource:

```console
$ git --no-pager grep cluster-monitoring-config ci-operator/step-registry/
ci-operator/step-registry/ipi/install/monitoringpvc/ipi-install-monitoringpvc-commands.sh:  name: cluster-monitoring-config
```

Can we do some `yq` merge magic so we don't have to worry about clobbering, as long as we keep the steps restricted to orthogonal config properties?

Reply:
Right, I'll rework this once https://github.com/openshift/release/pull/19545 lands